src/components/UI/ResourcesCard/ResourcesCard.tsx

```diff
@@ -0,0 +1,153 @@
+import styled from "@emotion/styled";
+import React, { FunctionComponent } from "react";
+import { Download } from "react-feather";
+import { mixin, vars } from "./../../../styles";
+
+interface ResourcesCardProps {
+  className?: string;
+  details: {
+    title: string;
+    youtubeEmbedCode?: string;
+    placeholderText?: string;
+    tag: string;
+    description: string;
+    downloads: {
+      fileName: string;
+      path: string;
+    }[];
+  };
+}
+
+const ResourcesCardUnStyled: FunctionComponent<ResourcesCardProps> = ({ className, details }) => {
+  const { placeholderText, youtubeEmbedCode, title, tag, description, downloads } = details;
+  const hasMedia = placeholderText || youtubeEmbedCode;
+
+  return (
+    <div className={`${className}`}>
+      {hasMedia && (
+        <div className="media-holder">
+          {youtubeEmbedCode ? (
+            <iframe
+              src={`https://www.youtube.com/embed/${youtubeEmbedCode}?rel=0`}
+              title={title}
+              frameBorder="0"
+              allowFullScreen
+              data-testid="youtubeEmbed-iframe"
+            />
+          ) : (
+            <div className="placehold" data-testid="placeholder">
+              <div className="d-flex justify-content-center align-items-center h-100">
+                <h5 className="flex mb-0">{placeholderText}</h5>
+              </div>
+            </div>
+          )}
+        </div>
+      )}
+      <div className="content">
+        <h3 className="title">
+          {youtubeEmbedCode ? (
+            <a
+              className="title"
+              href={`https://www.youtube.com/watch?v=${youtubeEmbedCode}`}
+              target="_blank"
+              rel="noopener noreferrer"
+              data-testid="youtubeEmbed-title-link"
+            >
+              {title}
+            </a>
+          ) : (
+            <span data-testid="placeholder-title">{title}</span>
+          )}
+        </h3>
+        {tag && <div className="tag">{tag}</div>}
+        <p>{description}</p>
+        <div className="py-2">
+          {downloads.map((download, index) => (
+            <a
+              className="font-weight-bold d-flex align-items-end mt-1"
+              href={download.path}
+              download={download.fileName}
+              key={index}
+              data-testid="download-link"
+            >
+              <Download />
+              <span className="ml-2">{download.fileName}</span>
+            </a>
+          ))}
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export const ResourcesCard = styled(ResourcesCardUnStyled)`
+  background-color: ${vars.white};
```

Comment:
wanna try this instead? it's something i realised recently:

`const ResourcesCardItem = styled.div`` `

then rename `const ResourcesCardUnStyled` to `export const ResourcesCard`

then use the styled component which u just defined, change `<div className={`${className}`}>` to `<ResourcesCardItem>`

doing the above steps can let you avoid always having to pass down `className?` props, which i find it super helpful

Reply:
ok